src/main/scala/edu/berkeley/ce/rockslicing/RockSlicer.scala

```diff
@@ -33,19 +32,44 @@ object RockSlicer {
       // Error message will already be printed out by InputProcessor
       System.exit(-1)
     }
-    val (globalOrigin, rockVolume, joints) = inputOpt.get
-    val starterBlocks = Seq(Block(globalOrigin, rockVolume))
+    val (globalOrigin, boundingBox, rockVolumeInputs, jointSetInputs) = inputOpt.get
+    val generatedInput = JointGenerator(globalOrigin, boundingBox, rockVolumeInputs, jointSetInputs)
+    val starterBlocks = Seq(Block(globalOrigin, generatedInput.rockVolume))
 
     // Generate a list of initial blocks before RDD-ifying it
-    val seedBlocks = if (arguments.numProcessors > 1) {
-      val processorJoints = LoadBalancer.generateProcessorJoints(starterBlocks.head, arguments.numProcessors)
-      processorJoints.foldLeft(starterBlocks) { (currentBlocks, joint) =>
-        currentBlocks.flatMap(_.cut(joint))
+    val seedJoints = if (arguments.numProcessors > 1) {
+      // Check if at least on of the input joint sets is persistent
+      val persistentCheck = jointSetInputs exists { jointSet =>
+        jointSet(3) == 100
+      }
+
+      if (!persistentCheck) {
+        println("ERROR: Input joint sets must contain at least one persistent joint set to run in parallel. Rerun" +
+          "analysis in serial if all joint sets are non-persistent.")
+        System.exit(-1)
+      }
+
+      val seedJoints = SeedJointSelector.searchJointSets(generatedInput.jointSets,
+        starterBlocks.head, arguments.numProcessors)
+
+      if (seedJoints.isEmpty) {
+        println("ERROR: Unable to find satisfactory seed joints")
+        System.exit(-1)
       }
+      seedJoints.get
```

Comment:
I've changed this to throw and exception earlier. There's definitely some abuse of returning `None` and catching it later instead of throwing an exception immediately, trying to eradicate that throughout.